src/main/java/org/wso2/carbon/inbound/cdc/CDCProcessor.java

```diff
@@ -0,0 +1,239 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.inbound.cdc;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.inbound.InboundProcessorParams;
+import org.apache.synapse.inbound.InboundTaskProcessor;
+import org.apache.synapse.task.TaskStartupObserver;
+import org.apache.synapse.util.resolver.SecureVaultResolver;
+import org.wso2.carbon.inbound.endpoint.common.InboundTask;
+import org.wso2.carbon.inbound.endpoint.protocol.PollingConstants;
+import org.wso2.carbon.inbound.endpoint.protocol.generic.GenericProcessor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_ALLOWED_OPERATIONS;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_DATABASE_ALLOW_PUBLIC_KEY_RETRIEVAL;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_DATABASE_PASSWORD;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_KEY_CONVERTER;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_KEY_CONVERTER_SCHEMAS_ENABLE;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_OFFSET_FLUSH_INTERVAL_MS;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_OFFSET_STORAGE;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_SCHEMA_HISTORY_INTERNAL;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_SCHEMA_HISTORY_INTERNAL_FILE_FILENAME;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_TOPIC_PREFIX;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_VALUE_CONVERTER;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_VALUE_CONVERTER_SCHEMAS_ENABLE;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.DEBEZIUM_SKIPPED_OPERATIONS;
+import static org.wso2.carbon.inbound.cdc.InboundCDCConstants.TRUE;
+
+public class CDCProcessor extends GenericProcessor implements TaskStartupObserver, InboundTaskProcessor {
+
+    private CDCPollingConsumer pollingConsumer;
+    private Properties cdcProperties;
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private static final String ENDPOINT_POSTFIX = "CDC" + COMMON_ENDPOINT_POSTFIX;
+    private static final String FILE_OFFSET_STORAGE_CLASS = "org.apache.kafka.connect.storage.FileOffsetBackingStore";
+    private static final String FILE_SCHEMA_HISTORY_STORAGE_CLASS = "io.debezium.storage.file.history.FileSchemaHistory";
```

Comment:
Seems these two constants appear in two places. Hence, shall we move them to the InboundCDCConstants class?

Reply:
Fixed